lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.rb

```diff
@@ -0,0 +1,222 @@
+          unless user_data_validation(csv_file_name, json_rows)
+            user_data_validation_outcome = false
+          end
+
+          # remove file extension
+          file_name = File.basename(csv_full_name, File.extname(csv_full_name))
+          json_objs[file_name] = json_rows
+        end
+      end
+      unless user_data_validation_outcome
+        terminate_prm_write_log('Error found in the user data. Check output log to see detail error messages', project_path, false)
+      end
+    end
+
+    # Make folder for json files; remove pre-existing first, if needed
+    json_path = "#{project_path}/user_data_json"
+    if !Dir.exist?(json_path)
+      Dir.mkdir(json_path)
+    else
+      FileUtils.rm_rf(json_path)
+      Dir.mkdir(json_path)
+    end
+
+    # Write all json files
+    json_objs.each do |file_name, json_rows|
+      json_obj = {}
+      json_obj[file_name] = json_rows
+      json_path_file = "#{json_path}/#{file_name}.json"
+      File.open(json_path_file, 'w:UTF-8') do |file|
+        file << JSON.pretty_generate(json_obj)
+      end
+    end
+
+    return json_path
+  end
+
+  # Load user data from project folder into standards database data structure
+  # Each user data object type is a new item in the @standards_data hash
+  # @author Doug Maddox, PNNL
+  # @param json_path [string] path to folder containing json files
+  def load_userdata_to_standards_database(json_path)
+    files = Dir.glob("#{json_path}/*.json").select { |e| File.file? e }
+    files.each do |file|
+      data = JSON.parse(File.read(file))
+      data.each_pair do |key, objs|
+        # Override the template in inherited files to match the instantiated template
+        if @standards_data[key].nil?
+          OpenStudio.logFree(OpenStudio::Debug, 'openstudio.standards.standard', "Adding #{key} from #{File.basename(file)}")
+        else
+          OpenStudio.logFree(OpenStudio::Debug, 'openstudio.standards.standard', "Overriding #{key} with #{File.basename(file)}")
+        end
+        @standards_data[key] = objs
+      end
+    end
+  end
+
+  # Perform user data preprocessing
+  # @param [CSV::ROW] row 2D array for each row.
+  def user_data_preprocessor(row)
+    new_array = []
+
+    # Strip the strings in the value
+    row.each do |sub_array|
+      new_array << sub_array.collect { |e| e ? e.strip : e }
+    end
+    # TODO: Future expansion can added to here.
+    # Convert the 2d array to hash
+    return new_array.to_h
+  end
+
+  # Perform user data validation
+  def user_data_validation(object_name, user_data)
+    # 1. Check user_spacetype and user_space LPD total % = 1.0
+    case object_name
+    when 'userdata_space', 'userdata_spacetype'
+      return check_userdata_space_and_spacetype(object_name, user_data)
+    when 'user_electric_equipment'
+      return check_userdata_electric_equipment(object_name, user_data)
+    else
+      return true
+    end
+  end
+
+  def check_userdata_electric_equipment(object_name, user_data)
```

Comment:
nice to add method documentation, but not required since it is an internal method